### PR TITLE
[doc] Release notes 10.0.0

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Updates the CHANGELOG and compatibility table for 10.0.0
## Related Issue

Closes PWA-1565

## Acceptance

@davemacaulay
@ericerway 
@tkacheva 

### Verification Stakeholders

@davemacaulay
@ericerway 
@tkacheva 

### Verification Steps

1. Open CHANGELOG
2. Verify content is accurate

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by jcalcaben** [comment]
Updated image

pwa-scaffold-template

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,34 +1,181 @@
...
+```sh
+yarn build:report
+```
+
+[![Image from Gyazo](https://i.gyazo.com/960b390007953778d67d298bd27b4886.png)](https://gyazo.com/960b390007953778d67d298bd27b4886)
+
+### Check or Money Order payments
+
+A PWA Studio extension that allows check or money order payments is now available thanks to the contribution of community member [Lars Roettig][].
+If the Magento backend has the **Check or Money Order** option enabled, this extension lets storefront customers use this payment option.
```
Comment (dpatil-magento) [inline CHANGELOG.md:34]: `Make check Payable to` and `Send Check to`  does not have GraphQL coverage, should this be noted in known issues or somewhere?  More info https://github.com/magento/pwa-studio/pull/2969#issuecomment-797654096

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,34 +1,181 @@
...
+| -------------------------------------------- | ----------- | --------- |
+| Removed auto-assign logic for new docs issue | **Update**  | [#3040][] |
+| Upgraded workbox to v6                       | **Update**  | [#2983][] |
+| Added coveralls badge back to the README     | **Update**  | [#2978][] |
+| Upgraded React to v17                        | **Update**  | [#2962][] |
+| Enabled singleton style tag                  | **Update**  | [#2906][] |
+| Updated prettier config                      | **Update**  | [#2900][] |
+| Removed cyclic dependencies                  | **Bugfix**  | [#2967][] |
 
 ## Known issues
```
Comment (dpatil-magento) [inline CHANGELOG.md:181]: GraphQL issue https://jira.corp.magento.com/browse/PWA-1298 is fixed on 2.4-develop, good to add this note/info info if customers like to get patch.

**Review by dpatil-magento — CHANGES_REQUESTED** [review]
Pls add https://github.com/magento/pwa-studio/pull/3065 merged on release branch.

**Review by dpatil-magento — APPROVED** [review]

**Review by davemacaulay — DISMISSED** [review]

**Review by supernova-at — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,45 +1,202 @@
...
+## What's new in 10.0.0
 
-PWA Studio 9.0.1 contains the same features, refactors, and various improvements in the previous [9.0.0 release][] plus a hotfix for the `create-pwa` CLI tool.
+PWA Studio 10.0.0 contains new features, refactors, bug fixes, and various improvements.
 
-[9.0.0 release]: https://github.com/magento/pwa-studio/releases/tag/v9.0.0
+### Build report tool
 
-### Hotfix for `create-pwa`
+The build report tool is a Buildpack CLI command that returns information about their storefront setup and development environment.
```
Comment (supernova-at) [inline CHANGELOG.md:20]:
```suggestion
The build report tool is a Buildpack CLI command that returns information about a storefront setup and development environment.
```

**Review by dpatil-magento — APPROVED** [review]

**Review by ericerway — DISMISSED** [review]

**Review by supernova-at — APPROVED** [review]

**Pull request merged**